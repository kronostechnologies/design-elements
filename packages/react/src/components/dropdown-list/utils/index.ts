export * from './dropdown-list-utils';
