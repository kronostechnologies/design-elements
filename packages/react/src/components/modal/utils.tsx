import React from 'react';
import styled from 'styled-components';
import { Button } from '../buttons/button';

const Title = styled.h2`
    font-size: 1.25rem;
    font-weight: var(--font-normal);
    height: 32px;
    line-height: 24px;
    margin-bottom: var(--spacing-2x);
    margin-top: 0;
`;

const Subtitle = styled.h3`
    font-size: 1rem;
    font-weight: var(--font-normal);
    height: 24px;
    line-height: 22px;
    margin: 0;
`;

const ConfirmButton = styled(Button)`
    margin-right: var(--spacing-1x);
`;

const CancelButton = styled(Button)``;

interface GetHeaderProps {
    titleId: string;
    title?: string;
    subtitle?: string;
}

interface GetFooterProps {
    closeModal(): void;
}

export const getHeader = ({ titleId, title, subtitle }: GetHeaderProps) => {
    if (title ||Â subtitle) {
        return (
            <>
                {title && <Title id={titleId} tabIndex={-1}>{title}</Title>}
                {subtitle && <Subtitle tabIndex={-1}>{subtitle}</Subtitle>}
            </>
        );
    } else return undefined;
};

export const getFooter = ({ closeModal }: GetFooterProps) => (
    <>
        <ConfirmButton label="Confirm" buttonType="primary"/>
        <CancelButton label="Cancel" buttonType="tertiary" onClick={closeModal}/>
    </>
);
