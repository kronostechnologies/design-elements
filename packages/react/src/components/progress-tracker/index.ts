export { ProgressTracker } from './progress-tracker';
export type { ProgressTrackerStep, ProgressTrackerProps } from './progress-tracker';
