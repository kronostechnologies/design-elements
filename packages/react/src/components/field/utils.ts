import { ReactNode } from 'react';
import { devConsole } from '../../utils/dev-console';
import { joinStrings } from '../../utils/string';

interface FieldIds {
    label?: string;
    hint?: string;
    invalid?: string;
}

export function getSlotId(prop: ReactNode, fieldId: string, propName: string): string | undefined {
    return prop ? `${fieldId}_${propName}` : undefined;
}

export function getSlotIds(
    fieldId: string,
    label?: ReactNode,
    hint?: ReactNode,
    invalid?: ReactNode,
): FieldIds {
    return {
        label: getSlotId(label, fieldId, 'label'),
        hint: getSlotId(hint, fieldId, 'hint'),
        invalid: getSlotId(invalid, fieldId, 'invalid'),
    };
}

export const ARIA_LABEL_WARNING = ''
    + 'label, aria-label and aria-labelledby are mutually exclusive properties. Only one should be provided.';

export const validateAriaLabels = (
    label?: string,
    ariaLabel?: string,
    ariaLabelledby?: string,
): boolean => {
    const labelCount = [!!label, !!ariaLabel, !!ariaLabelledby].filter(Boolean).length;
    if (labelCount === 0 || labelCount > 1) {
        devConsole.warn(ARIA_LABEL_WARNING);
        return false;
    }
    return true;
};

export function getAriaLabel(
    label?: string,
    ariaLabel?: string,
    ariaLabelledby?: string,
): string | undefined {
    return validateAriaLabels(label, ariaLabel, ariaLabelledby) ? ariaLabel : undefined;
}

export function getAriaLabelledby(
    { label }: FieldIds,
    additionalLabelledby?: string,
): string | undefined {
    return joinStrings(label, additionalLabelledby);
}

export function getAriaDescribedby(
    { invalid, hint }: FieldIds,
    additionalDescribedby?: string,
): string | undefined {
    return joinStrings(invalid, hint, additionalDescribedby);
}
