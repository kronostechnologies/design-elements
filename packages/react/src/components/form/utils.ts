import { devConsole } from '../../utils/dev-console';

interface FormFieldIds {
    label?: string;
    hint?: string;
    invalid?: string;
}

function makeSlotId(prop: React.ReactNode, formFieldId: string, propName: string): string | undefined {
    if (!prop) {
        return undefined;
    }

    return `${formFieldId}_${propName}`;
}

export function getSlotIds(
    formFieldId: string,
    label?: React.ReactNode,
    hint?: React.ReactNode,
    invalid?: React.ReactNode,
): FormFieldIds {
    return {
        label: makeSlotId(label, formFieldId, 'label'),
        hint: makeSlotId(hint, formFieldId, 'hint'),
        invalid: makeSlotId(invalid, formFieldId, 'invalid'),
    };
}

export const ARIA_LABEL_WARNING = ''
    + 'label, aria-label and aria-labelledby are mutually exclusive properties. Only one should be provided.';

export const validateAriaLabels = (
    label?: string,
    ariaLabel?: string,
    ariaLabelledby?: string,
): boolean => {
    const labelCount = [!!label, !!ariaLabel, !!ariaLabelledby].filter(Boolean).length;
    if (labelCount === 0 || labelCount > 1) {
        devConsole.warn(ARIA_LABEL_WARNING);
        return false;
    }
    return true;
};

export function joinStrings(...strings: Array<string | undefined>): string | undefined {
    const filteredStrings = strings.filter(Boolean);
    return filteredStrings.length > 0 ? filteredStrings.join(' ') : undefined;
}

export function getAriaLabel(
    label?: string,
    ariaLabel?: string,
    ariaLabelledby?: string,
): string | undefined {
    return validateAriaLabels(label, ariaLabel, ariaLabelledby) ? ariaLabel : undefined;
}

export function getAriaLabelledBy(
    { label }: FormFieldIds,
    additionalLabelledBy?: string,
): string | undefined {
    return joinStrings(label, additionalLabelledBy);
}

export function getAriaDescribedBy(
    { invalid, hint }: FormFieldIds,
    additionalDescribedBy?: string,
): string | undefined {
    return joinStrings(invalid, hint, additionalDescribedBy);
}
