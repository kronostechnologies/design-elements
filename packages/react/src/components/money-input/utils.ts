import { formatCurrency } from '../../utils/currency';

export function safeFormatCurrency(
    value: number | null,
    precision: number,
    locale: string,
    currency: string,
): string {
    return value === null ? '' : formatCurrency(value, precision, locale, currency);
}

export function roundValueToPrecision(amount: number | null, precision: number): number | null {
    return amount !== null && !Number.isNaN(amount)
        ? Math.round(amount * (10 ** precision)) / (10 ** precision)
        : null;
}

export function parseAndRound(val: string, precision: number): number | null {
    return val === '' ? null : roundValueToPrecision(Number(val.replace(',', '.')), precision);
}
