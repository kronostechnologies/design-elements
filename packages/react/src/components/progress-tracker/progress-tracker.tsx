import { type FC, MouseEvent, ReactElement } from 'react';
import styled, { css } from 'styled-components';
import { useTranslation } from '../../i18n/use-translation';
import { clamp } from '../../utils/math';
import { Icon } from '../icon';
import { ScreenReaderOnlyText } from '../screen-reader-only-text';

const Container = styled.section`
    padding: var(--spacing-2x) 0;
    position: relative;
    width: 100%;
`;

const Steps = styled.ol`
    align-items: start;
    counter-reset: step;
    display: grid;
    grid-auto-columns: unset;
    grid-gap: 1rem;
    grid-template-columns: repeat(auto-fit, minmax(1rem, 1fr));
    margin: 0;
    padding: 0;
`;

const Label = styled.span`
    font-size: 0.875rem;
    font-weight: var(--font-normal);
    letter-spacing: 0.0125rem;
    line-height: 1.25rem;
    margin-top: var(--spacing-half);
`;

const StepLink = styled.a`
    display: inline-block;
    margin-top: calc(var(--size-2x) * -1);
    padding-top: var(--size-2x);
    text-decoration: none;
`;

const StyledStep = styled.li<{ $linear: boolean }>`
    align-items: center;
    color: ${({ theme }) => theme.component['progress-tracker-step-todo-text-color']};
    display: flex;
    flex-direction: column;
    font-weight: var(--font-normal);
    justify-content: flex-end;
    position: relative;
    text-align: center;

    &::before {
        align-items: center;
        background-color: ${({ theme }) => theme.component['progress-tracker-step-todo-background-color']};
        border: 0.125rem solid;
        border-radius: 50%;
        box-sizing: border-box;
        color: #1b1c1e;
        content: counter(step);
        counter-increment: step;
        display: flex;
        font-size: 0.75rem;
        height: var(--size-1halfx);
        justify-content: center;
        margin: 0 auto 0.25rem;
        text-align: center;
        width: var(--size-1halfx);
    }

    &::after {
        background-color: ${({ theme }) => theme.component['progress-tracker-bridge-empty-background-color']};
        content: '';
        height: 0.25rem;
        left: calc(-50% - 0.5rem);
        position: absolute;
        top: 0.625rem;
        width: 100%;
        z-index: -1;
    }

    &:first-child::after {
        content: none;
    }
`;

const CompletedStep = styled(StyledStep)`
    &::before {
        background-color: ${({ theme }) => theme.component['progress-tracker-step-completed-background-color']};
        border-color: ${({ theme }) => theme.component['progress-tracker-step-completed-border-color']};
        color: ${({ theme }) => theme.component['progress-tracker-step-completed-text-color']};
        font-weight: var(--font-semi-bold);
    }

    &::after {
        ${({ $linear, theme }) => $linear && css`
            background-color: ${theme.component['progress-tracker-bridge-filled-background-color']};
        `}
    }

    ${Label} {
        color: ${({ theme }) => theme.component['progress-tracker-step-completed-label-text-color']};
    }
`;

const ActiveStep = styled(StyledStep)`
    &::before {
        border-color: ${({ theme }) => theme.component['progress-tracker-step-current-border-color']};
        border-width: 0.25rem;
        color: ${({ theme }) => theme.component['progress-tracker-step-current-text-color']};
        font-weight: var(--font-semi-bold);
        height: var(--size-2x);
        margin: -0.25rem auto 0;
        width: var(--size-2x);
    }

    &::after {
        ${({ $linear, theme }) => $linear && css`
            background-color: ${theme.component['progress-tracker-bridge-filled-background-color']};
        `}
    }

    ${Label} {
        color: ${({ theme }) => theme.component['progress-tracker-step-current-label-text-color']};
        font-weight: var(--font-semi-bold);
    }
`;

const UncompletedStep = styled(StyledStep)`
    &::before {
        border-color: ${({ theme }) => theme.component['progress-tracker-step-uncompleted-border-color']};
        color: ${({ theme }) => theme.component['progress-tracker-step-uncompleted-text-color']};
    }

    ${Label} {
        color: ${({ theme }) => theme.component['progress-tracker-step-uncompleted-label-text-color']};
    }
`;

const NotificationBadgeIcon = styled(Icon)`
    color: ${({ theme }) => theme.component['progress-tracker-notification-badge-color']};
    fill: ${({ theme }) => theme.component['progress-tracker-notification-badge-fill-color']};
    left: calc(50% + 0.25rem);
    position: absolute;
    top: -0.5rem;
`;

export interface ProgressTrackerStep {
    href?: string;
    label: string;
    completion?: 'completed' | 'uncompleted';
    onClick?: (stepNumber: number) => void;
}

interface StepProps {
    step: ProgressTrackerStep,
    stepNumber: number,
    value: number,
    linear: boolean
}

const Step: FC<StepProps> = ({
    step, stepNumber, value, linear,
}): ReactElement => {
    const { t } = useTranslation('progress-tracker');

    let StepComponent: typeof StyledStep;
    let dataTestId: string | undefined;
    let screenReaderText: string | undefined;
    const isLink = step.href || step.onClick;
    const showUncompletedIcon = !linear && step.completion === 'uncompleted';

    if (stepNumber === value) {
        dataTestId = 'progress-tracker-step-current';
        StepComponent = ActiveStep;
    } else if ((linear && stepNumber < value) || (!linear && step.completion === 'completed')) {
        dataTestId = 'progress-tracker-step-completed';
        screenReaderText = t('completedAriaLabel');
        StepComponent = CompletedStep;
    } else {
        dataTestId = 'progress-tracker-step-uncompleted';
        screenReaderText = t('uncompletedAriaLabel');
        StepComponent = UncompletedStep;
    }

    const content = (
        <>
            {showUncompletedIcon && <NotificationBadgeIcon name='alertCircle' size='16' aria-hidden="true" />}
            {step.label && <Label data-testid="progress-tracker-label">{step.label}</Label>}
            {screenReaderText && <ScreenReaderOnlyText label={screenReaderText} />}
        </>
    );

    const linkClickHandler = (event: MouseEvent<HTMLAnchorElement>): void => {
        if (!step.href) {
            event.preventDefault();
        }
        step.onClick?.(stepNumber);
    };

    return (
        <StepComponent
            key={stepNumber}
            data-testid={dataTestId}
            aria-current={stepNumber === value ? 'step' : undefined}
            $linear={linear}
        >
            {isLink ? (
                <StepLink href={step.href ?? '#'} onClick={step.onClick && linkClickHandler}>
                    {content}
                </StepLink>
            ) : (
                content
            )}
        </StepComponent>
    );
};

export interface ProgressTrackerProps {
    ariaLabel?: string;
    className?: string;
    linear?: boolean;
    steps: ProgressTrackerStep[];
    value: number;
}

export const ProgressTracker: FC<ProgressTrackerProps> = ({
    ariaLabel,
    linear = true,
    className,
    steps,
    value,
}) => {
    const max = steps.length;
    const clampValue = clamp(value, 1, max);
    const hasAnyLink = steps.some((step) => step.href || step.onClick);

    if (!ariaLabel) {
        console.warn('ariaLabel is required for Accessibility');
    }

    return (
        <Container className={className} aria-label={ariaLabel} as={hasAnyLink ? 'nav' : undefined}>
            <Steps data-testid="progress-tracker">
                {steps.map((step, index) => (
                    <Step
                        step={step}
                        stepNumber={index + 1}
                        value={clampValue}
                        linear={linear}
                        // eslint-disable-next-line react/no-array-index-key
                        key={index}
                    />
                ))}
            </Steps>
        </Container>
    );
};
