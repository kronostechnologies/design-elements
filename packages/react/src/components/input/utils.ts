import { InputFieldProps, InputProps } from './types';
import { FieldContainerProps } from '../field/types';

interface ParsedInputFieldProps {
    inputProps: InputProps;
    fieldControlledProps: FieldContainerProps;
}

export function parseInputFieldProps(
    props: InputFieldProps,
): ParsedInputFieldProps {
    const {
        ariaLabel,
        ariaLabelledby,
        ariaDescribedby,
        valid,
        id,
        disabled,
        required,
        className,
        noMargin,
        label,
        hint,
        validationErrorMessage,
        noInvalidFieldIcon,
        tooltip,
        ...inputProps
    } = props;

    const fieldControlledProps: FieldContainerProps = {
        id,
        ariaLabel,
        ariaLabelledby,
        ariaDescribedby,
        valid,
        className,
        noMargin,
        label,
        hint,
        validationErrorMessage,
        noInvalidFieldIcon,
        tooltip,
        disabled,
        required,
    };

    return { inputProps, fieldControlledProps };
}
