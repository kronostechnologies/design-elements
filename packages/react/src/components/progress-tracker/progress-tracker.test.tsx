import { shallow } from 'enzyme';
import { findByTestId, getByTestId } from '../../test-utils/enzyme-selectors';
import { renderWithProviders } from '../../test-utils/renderer';
import { range } from '../../utils/range';
import { ProgressTracker, ProgressTrackerStep } from './progress-tracker';

function generateSteps(count: number): ProgressTrackerStep[] {
    return range(1, count).map((i) => ({ label: `Step ${i}` }));
}

describe('Progress Component', () => {
    it('should render all steps', () => {
        const steps = generateSteps(3);

        const wrapper = shallow(<ProgressTracker steps={steps} value={1} />);
        const stepsWrapper = getByTestId(wrapper, 'progress-tracker');

        const allSteps = findByTestId(stepsWrapper, 'progress-tracker-step-', '^');
        expect(allSteps.length).toBe(3);
    });

    it('should render all steps labels', () => {
        const steps = generateSteps(3);

        const wrapper = shallow(<ProgressTracker steps={steps} value={1} />);
        const stepsWrapper = getByTestId(wrapper, 'progress-tracker');

        const allStepsLabels = findByTestId(stepsWrapper, 'progress-tracker-step-', '^')
            .map((w) => findByTestId(w, 'progress-tracker-label').text());
        expect(allStepsLabels).toEqual(expect.arrayContaining(['Step 1', 'Step 2', 'Step 3']));
    });

    it('should have completed steps', () => {
        const steps = generateSteps(3);

        const wrapper = shallow(<ProgressTracker steps={steps} value={3} />);
        const stepsWrapper = getByTestId(wrapper, 'progress-tracker');

        const completedSteps = findByTestId(stepsWrapper, 'progress-tracker-step-completed')
            .map((w) => findByTestId(w, 'progress-tracker-label').text());
        expect(completedSteps).toEqual(expect.arrayContaining(['Step 1', 'Step 2']));
    });

    it('should have current step', () => {
        const steps = generateSteps(3);

        const wrapper = shallow(<ProgressTracker steps={steps} value={2} />);
        const stepsWrapper = getByTestId(wrapper, 'progress-tracker');

        const currentStep = findByTestId(stepsWrapper, 'progress-tracker-step-current')
            .map((w) => findByTestId(w, 'progress-tracker-label').text());
        expect(currentStep).toEqual(expect.arrayContaining(['Step 2']));
    });

    it('should fallback to first step if value is less than 1', () => {
        const steps = generateSteps(3);

        const wrapper = shallow(<ProgressTracker steps={steps} value={0} />);
        const stepsWrapper = getByTestId(wrapper, 'progress-tracker');

        const currentStep = findByTestId(stepsWrapper, 'progress-tracker-step-current')
            .map((w) => findByTestId(w, 'progress-tracker-label').text());
        expect(currentStep).toEqual(expect.arrayContaining(['Step 1']));
    });

    it('should fallback to last step if value is higher than number of steps', () => {
        const steps = generateSteps(3);

        const wrapper = shallow(<ProgressTracker steps={steps} value={4} />);
        const stepsWrapper = getByTestId(wrapper, 'progress-tracker');

        const currentStep = findByTestId(stepsWrapper, 'progress-tracker-step-current')
            .map((w) => findByTestId(w, 'progress-tracker-label').text());
        expect(currentStep).toEqual(expect.arrayContaining(['Step 3']));
    });

    it('should have uncompleted steps', () => {
        const steps = generateSteps(3);

        const wrapper = shallow(<ProgressTracker steps={steps} value={1} />);
        const stepsWrapper = getByTestId(wrapper, 'progress-tracker');

        const uncompletedStep = findByTestId(stepsWrapper, 'progress-tracker-step-uncompleted')
            .map((w) => findByTestId(w, 'progress-tracker-label').text());
        expect(uncompletedStep).toEqual(expect.arrayContaining(['Step 2', 'Step 3']));
    });

    describe('Snapshots', () => {
        it('with labels', () => {
            const steps = generateSteps(3);

            const wrapper = renderWithProviders(<ProgressTracker steps={steps} value={2} />);

            expect(wrapper).toMatchSnapshot();
        });

        it('without labels', () => {
            const steps: ProgressTrackerStep[] = [{}, {}, {}];

            const wrapper = renderWithProviders(<ProgressTracker steps={steps} value={2} />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
