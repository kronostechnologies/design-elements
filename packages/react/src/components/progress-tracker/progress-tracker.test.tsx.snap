// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Component Snapshots with labels 1`] = `
.c6 {
  border: 0;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c0 {
  padding: var(--spacing-2x) 0;
  position: relative;
  width: 100%;
}

.c1 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  counter-reset: step;
  display: grid;
  grid-auto-columns: unset;
  grid-gap: 1rem;
  grid-template-columns: repeat(auto-fit,minmax(1rem,1fr));
  margin: 0;
  padding: 0;
}

.c5 {
  font-size: 0.875rem;
  font-weight: var(--font-normal);
  -webkit-letter-spacing: 0.0125rem;
  -moz-letter-spacing: 0.0125rem;
  -ms-letter-spacing: 0.0125rem;
  letter-spacing: 0.0125rem;
  line-height: 1.25rem;
  margin-top: var(--spacing-half);
}

.c2 {
  color: #004E78;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-weight: var(--font-normal);
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  position: relative;
  text-align: center;
}

.c2::before {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFFFFF;
  border: 0.125rem solid;
  border-radius: 50%;
  box-sizing: border-box;
  color: #1b1c1e;
  content: counter(step);
  counter-increment: step;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0.75rem;
  height: var(--size-1halfx);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 auto 0.25rem;
  text-align: center;
  width: var(--size-1halfx);
}

.c2::after {
  background-color: #DBDEE1;
  content: '';
  height: 0.25rem;
  left: calc(-50% - 0.5rem);
  position: absolute;
  top: 0.625rem;
  width: 100%;
  z-index: -1;
}

.c2:first-child::after {
  content: none;
}

.c3::before {
  background-color: #004E78;
  border-color: #004E78;
  color: #FFFFFF;
  font-weight: var(--font-bold);
}

.c3::after {
  background-color: #004E78;
}

.c3 .c4 {
  color: #006296;
}

.c7::before {
  border-color: #004E78;
  border-width: 0.25rem;
  color: #003A5A;
  font-weight: var(--font-bold);
  height: var(--size-2x);
  margin: -0.25rem auto 0;
  width: var(--size-2x);
}

.c7::after {
  background-color: #004E78;
}

.c7 .c4 {
  color: #003A5A;
  font-weight: var(--font-bold);
}

.c8::before {
  border-color: #B7BBC2;
  color: #1B1C1E;
}

.c8 .c4 {
  color: #60666E;
}

<section
  class="c0"
>
  <ol
    class="c1"
    data-testid="progress-tracker"
  >
    <li
      class="c2 c3"
      data-testid="progress-tracker-step-completed"
    >
      <span
        class="c4 c5"
        data-testid="progress-tracker-label"
      >
        Step 1
      </span>
      <span
        class="c6"
      >
        completed
      </span>
    </li>
    <li
      aria-current="step"
      class="c2 c7"
      data-testid="progress-tracker-step-current"
    >
      <span
        class="c4 c5"
        data-testid="progress-tracker-label"
      >
        Step 2
      </span>
    </li>
    <li
      class="c2 c8"
      data-testid="progress-tracker-step-uncompleted"
    >
      <span
        class="c4 c5"
        data-testid="progress-tracker-label"
      >
        Step 3
      </span>
      <span
        class="c6"
      >
        not completed
      </span>
    </li>
  </ol>
</section>
`;

exports[`Progress Component Snapshots without labels 1`] = `
.c4 {
  border: 0;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c0 {
  padding: var(--spacing-2x) 0;
  position: relative;
  width: 100%;
}

.c1 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  counter-reset: step;
  display: grid;
  grid-auto-columns: unset;
  grid-gap: 1rem;
  grid-template-columns: repeat(auto-fit,minmax(1rem,1fr));
  margin: 0;
  padding: 0;
}

.c2 {
  color: #004E78;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-weight: var(--font-normal);
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  position: relative;
  text-align: center;
}

.c2::before {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFFFFF;
  border: 0.125rem solid;
  border-radius: 50%;
  box-sizing: border-box;
  color: #1b1c1e;
  content: counter(step);
  counter-increment: step;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0.75rem;
  height: var(--size-1halfx);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 auto 0.25rem;
  text-align: center;
  width: var(--size-1halfx);
}

.c2::after {
  background-color: #DBDEE1;
  content: '';
  height: 0.25rem;
  left: calc(-50% - 0.5rem);
  position: absolute;
  top: 0.625rem;
  width: 100%;
  z-index: -1;
}

.c2:first-child::after {
  content: none;
}

.c3::before {
  background-color: #004E78;
  border-color: #004E78;
  color: #FFFFFF;
  font-weight: var(--font-bold);
}

.c3::after {
  background-color: #004E78;
}

.c5::before {
  border-color: #004E78;
  border-width: 0.25rem;
  color: #003A5A;
  font-weight: var(--font-bold);
  height: var(--size-2x);
  margin: -0.25rem auto 0;
  width: var(--size-2x);
}

.c5::after {
  background-color: #004E78;
}

.c6::before {
  border-color: #B7BBC2;
  color: #1B1C1E;
}

<section
  class="c0"
>
  <ol
    class="c1"
    data-testid="progress-tracker"
  >
    <li
      class="c2 c3"
      data-testid="progress-tracker-step-completed"
    >
      <span
        class="c4"
      >
        completed
      </span>
    </li>
    <li
      aria-current="step"
      class="c2 c5"
      data-testid="progress-tracker-step-current"
    />
    <li
      class="c2 c6"
      data-testid="progress-tracker-step-uncompleted"
    >
      <span
        class="c4"
      >
        not completed
      </span>
    </li>
  </ol>
</section>
`;
